### ISSUE TITLE
P1 experimental pitch: add PESTO and torchcrepe as alternate Pitch Contour engines

### ISSUE THREAD

**Issue by gr-rr** [ListenCloser/listencloser#1236]
Parent pitch capability: #1087. Active first implementation: PR #1216. Experimental interpretation UX: #1173. Representation authority: #613. General transcription remains #931.

## Product problem

Pitch Contour should be a stable musical object with replaceable F0 engines behind it. PR #1216 is currently landing the first bounded `pYIN` implementation and the synchronized auxiliary lane. Once that seam is on `main`, add maintained learned pitch estimators as explicit alternate interpretations without changing the product concept.

This issue does **not** reopen the Pitch Contour UI or block #1216. Backend/adapter work should stack on the landed #1216 contract; do not edit the active PR branch unless its owner explicitly adopts this scope.

### torchcrepe

Official repo: https://github.com/maxrmorrison/torchcrepe

Observed upstream facts at issue creation:
- MIT licensed;
- PyTorch CREPE implementation;
- package version `0.0.24` in current setup metadata;
- small dependency surface: `torch`, `torchaudio`, `librosa`, `scipy`, `resampy`, `tqdm`;
- therefore a strong candidate for the existing Torch 2.6 worker without a separate ML runtime.

### PESTO

Official repo: https://github.com/SonyCSLParis/pesto

Observed upstream facts at issue creation:
- lightweight PyTorch/Torchaudio inference package (`pesto-pitch`);
- current inference repo is LGPL-3.0;
- maintained inference API supports model reuse/batching and outputs pitch + method-specific confidence;
- upstream reports substantially faster-than-real-time CPU inference for its lightweight model family;
- exact released model/checkpoint asset terms must be verified and recorded separately from code license.

Do not treat upstream speed/accuracy claims as local product evidence.

## Runtime shape

After #1216 lands, preserve one normalized Pitch Contour result contract:

```text
exact source audio Version
  -> selected pitch engine
  -> frames: time + F0 Hz / unvoiced + engine-native score where meaningful
  -> durable experimental pitch-contour report
  -> existing synchronized Pitch Contour lane
```

If #1216 does not introduce an explicit pitch-engine protocol, add only the smallest local typed seam needed to avoid three separate capability implementations. Do not create a generic plugin/model framework.

## Backend slice

Add explicit engine identities for:
- current `pyin` control/default for the experimental capability;
- `torchcrepe`;
- `pesto`.

Requirements:
1. all consume the same exact source audio Version and declare preprocessing/sample-rate/hop semantics;
2. normalize into the existing contour report without coercing F0 into MIDI note events;
3. retain each engine's actual voicing/confidence semantics rather than relabeling scores as generic correctness confidence;
4. pin package/model identity and record model/checkpoint hashes where applicable;
5. heavy/model imports stay lazy;
6. one selected-engine failure remains capability-local and never silently substitutes pYIN/another engine;
7. preserve prior generated contour interpretations as independent immutable reports;
8. no automatic winner/default change in this issue.

Prefer direct installation into the existing worker where dependency resolution remains clean. If PESTO's LGPL or model-asset packaging requires a materially different distribution boundary, document the exact constraint and keep that engine behind the appropriate optional runtime rather than weakening the worker contract.

## Minor UI surfacing

Reuse the existing #1216 Pitch Contour lane and #1173 progressive-disclosure grammar only:

```text
Pitch Contour · Experimental
  Details
    Interpretation · pYIN
    Try another interpretation
      PESTO
      torchcrepe
```

Rules:
- engine names remain secondary vocabulary under Details, not a primary Add-analysis chooser;
- generating an alternate preserves the existing result;
- selecting an already-generated contour is cheap/reversible and keeps shared playhead/selection;
- no side-by-side pitch dashboard or new top-level tab;
- do not show fake global confidence comparisons between engines.

## Product proof now; canonize later

Use a few lawful examples:
- clean singing/monophonic voice;
- expressive instrument with vibrato/bends;
- one weak/noisy voice example;
- one dense/polyphonic negative control.

Record for each engine:
- obvious octave/subharmonic/voicing failure behavior by listening/inspection;
- warm/cold runtime and peak RSS;
- result density/coverage;
- whether switching interpretations is genuinely useful in the existing lane.

This is runtime/product inspection. A trusted/default F0 decision is a later focused evaluation only when real use creates that decision.

## Acceptance

- [ ] PR #1216 is merged or its exact resulting pitch-report contract is otherwise stable before conflicting integration work lands;
- [ ] torchcrepe runs through the same Pitch Contour backend path on a real Work;
- [ ] PESTO runs through the same path, or an exact license/model/runtime blocker is recorded without contaminating the main worker;
- [ ] pYIN remains the current control/default for the experimental capability until separately changed;
- [ ] exact engine/model/preprocessing/source provenance persists for every result;
- [ ] previous contour interpretations are preserved rather than overwritten;
- [ ] `Details -> Try another interpretation` can generate/select the alternates in the existing lane;
- [ ] no new top-level capability/UI or generic engine framework is introduced;
- [ ] several materially different Works are inspected and latency/RSS recorded;
- [ ] relevant backend/browser/real-stack gates are green.

Refs #1087 #1216 #1173 #613 #931 #498.

**Comment by gr-rr** [comment]
### Feasibility note from upstream inspection

PESTO looks especially cheap operationally:
- the inference repository itself contains the pretrained `mir-1k.ckpt` (~124 KB) and `mir-1k_g7.ckpt` (~535 KB) assets under `pesto/weights/`;
- code/repo license is LGPL-3.0;
- runtime API is ordinary PyTorch/Torchaudio and supports loading the model once per process.

So the default implementation attempt should be **direct worker integration**, not a sidecar. Verify there is no separate checkpoint notice overriding the repository license, then pin/checksum the exact bundled weight.

torchcrepe remains the clean permissive control (MIT; Torch/Torchaudio/librosa/scipy stack). Start with these two direct adapters after #1216 lands; only introduce runtime isolation if an actual dependency/license conflict appears.
